src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Collections;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.IO.Ports;
+using System.Net.Sockets;
+
+namespace System.IO.Ports
+{
+    internal sealed partial class SerialStream : Stream
+    {
+        private const int MaxDataBits = 8;
+        private const int MinDataBits = 5;
+
+        // members supporting properties exposed to SerialPort
+        private string _portName;
+        private bool _inBreak = false;
+        private Handshake _handshake;
+
+        private SafeFileHandle _handle = null;
+
+        // three different events, also wrapped by SerialPort.
+        // called when one character is received.
+        internal event SerialDataReceivedEventHandler DataReceived;
+
+#pragma warning disable CS0067 // Events shared by Windows and Linux, on Linux we currently never call them
+        // called when any of the pin/ring-related triggers occurs
+        internal event SerialPinChangedEventHandler PinChanged;
+        // called when any runtime error occurs on the port (frame, overrun, parity, etc.)
+        internal event SerialErrorReceivedEventHandler ErrorReceived;
+#pragma warning restore CS0067
+
+        // ----SECTION: inherited properties from Stream class ------------*
+
+        // These six properites are required for SerialStream to inherit from the abstract Stream class.
+        // Note four of them are always true or false, and two of them throw exceptions, so these
+        // are not usefully queried by applications which know they have a SerialStream, etc...
+        public override bool CanRead
+        {
+            get { return (_handle != null); }
+        }
+
+        public override bool CanSeek
+        {
+            get { return false; }
+        }
+
+        public override bool CanTimeout
+        {
+            get { return (_handle != null); }
```

Comment:
>return (_handle != null) [](start = 18, length = 24)

would this depends on ReadTimeout too?

Reply:
I think this is about the capability rather than the current setting but will validate if this matches Windows and docs

Reply:
Actually this must match Windows - considering I moved only the same code to the shared file :-) the doc does not go into details if this is in general a capability or if in the current state this can happen. I think we should leave it as is to match Windows even though checking for infinite timeout makes sense